src/main/java/com/example/everkan/security/jwt/filter/AuthTokenFilter.java

```diff
@@ -0,0 +1,66 @@
+package com.example.everkan.security.jwt.filter;
+
+import com.example.everkan.appuser.AppUserDetailsServiceImpl;
+import com.example.everkan.security.jwt.JwtSecurityConstants;
+import com.example.everkan.security.jwt.JwtUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.lang.NonNull;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.web.authentication.WebAuthenticationDetailsSource;
+import org.springframework.util.StringUtils;
+import org.springframework.web.filter.OncePerRequestFilter;
+
+import javax.servlet.FilterChain;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+
+
+public class AuthTokenFilter extends OncePerRequestFilter {
+
+    private static final Logger logger = LoggerFactory.getLogger(AuthTokenFilter.class);
+
+    @Autowired
+    private  JwtUtils jwtUtils;
+
+    @Autowired
+    private  AppUserDetailsServiceImpl appUserDetailsService;
```

Comment:
Autowiring into properties usually makes classes hard to test.

You could add these attributes to a constructor to achieve the same effect and simplify testing.

Reply:
You are right. It does make testing more difficult. I removed the annotations and pass the objects through the constructor instead.

Reply:
The alternative is to use reflection utils (e.g. `org.springframework.util.ReflectionUtils`), but usually that makes it more complicated than necessary.